File: clients/dart/lib/model/role_mapping.dart
part of api;


@Entity()
class RoleMapping {
  
  num id = null;
  
  /* The principal type, such as user, application, or role */
  String principalType = null;
  
  
  String principalId = null;
  
  
  num roleId = null;
  
  
  RoleMapping();

  @override
  String toString()  {
    return 'RoleMapping[id=$id, principalType=$principalType, principalId=$principalId, roleId=$roleId, ]';
  }

}

File: clients/typescript-angular/API/Client/RoleMapping.ts
/// <reference path="api.d.ts" />

namespace API.Client {
    'use strict';

    /**
     * Map principals to roles
     */
    export interface RoleMapping {

        id?: number;

        /**
         * The principal type, such as user, application, or role
         */
        principalType?: string;

        principalId?: string;

        roleId?: number;
    }

}
